File: Data Structures/Hash Maps/HashMaps.cpp
#include <iostream>
#include <unordered_map>
using namespace std;

int main(){
    unordered_map <char, int> hashmap;
    // insert
    hashmap['a'] = 1;
    hashmap['b'] = 2;
    hashmap['c'] = 3;

    //traverse
    for(const auto &temp: hashmap){
        cout << temp.first <<":" <<temp.second<<endl;
    }

    //search
    if(hashmap.find('b') != hashmap.end()){
        cout<<hashmap['b'];
    }
    
    //delete
    hashmap.erase('b');

    //traverse
    for(const auto &temp: hashmap){
        cout << temp.first <<":" <<temp.second<<endl;
    }
    
    return 0 ;
}

File: Data Structures/Hash Maps/HashMaps.java
import java.util.HashMap;
import java.util.Map;

public class HashMaps {
    public static void main(String[] args){
        HashMap <Integer, String> hashmap = new HashMap<>();
        //insert
        hashmap.put(1, "a");
        hashmap.put(2, "b");
        hashmap.put(3, "c");

        //travese
        for(Map.Entry<Integer, String> temp: hashmap.entrySet()){
            System.out.printf("%d: %s",temp.getKey(),temp.getValue());
        }

        //search
        if(hashmap.containsKey(2)){
            hashmap.remove(2);
        }

        //travese
        for(Map.Entry<Integer, String> temp: hashmap.entrySet()){
            System.out.printf("%d: %s",temp.getKey(),temp.getValue());
        }  
    }
}
